### PR TITLE
General design

### LINKED ISSUE #22: HTML-CSS: General design

### PR DESCRIPTION
**Pull request by Blatoy**
Close #22 
- Add SPA navigation
- Add auth "middleware"

### CODE REVIEW

File: marcdown_project/frontend/src/components/common/Nav.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <nav>
+          <NavItem url="/" text="" exact={true} >
+            <img src="/static/images/logo.png" alt="" />
+          </NavItem>
+          <ul>
+            <li>
+              <NavItem url="/" exact={true}>Home</NavItem>
+            </li>
+            <li>
+              <NavItem url="/editor">Editor</NavItem>
```
Comment (g-wizzy) [inline marcdown_project/frontend/src/components/common/Nav.js]: How is the editor linked in the navigation bar ?

File: marcdown_project/frontend/src/pages/Home.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          <div>
+            {/* tag list */}
+          </div>
+        </div>
+        <div>
+          <h1>My notes</h1>
+          <div></div>
+          <h1>Shared with me</h1>
+          <div></div>
+        </div>
```
Comment (g-wizzy) [inline marcdown_project/frontend/src/pages/Home.js:19]: Could add a "favorites" line

Comment (Blatoy) [reply marcdown_project/frontend/src/pages/Home.js:19]: That's temporary text anyway, will be changed when working on the page itself

File: marcdown_project/frontend/src/components/common/Nav.js (hunk 1 above)
Comment (Blatoy) [reply marcdown_project/frontend/src/components/common/Nav.js]: Fixed it, replaced by /note/:id

**Review by g-wizzy — APPROVED** [review]

**Pull request merged**